src/app/webbrowser/TabComponent.qml

```diff
@@ -363,12 +364,18 @@ Component {
                     }
                 }
             }
+            */
             onIconChanged: {
+
+                if (webviewimpl.incognito) {
+                    return
+                }
+
                 if (webviewInternal.storedUrl.toString()) {
-                    HistoryModel.update(webviewInternal.storedUrl, webviewInternal.title, icon)
+                    HistoryModel.update(webviewInternal.storedUrl, webviewInternal.title, (icon.toString().substring(0,16) == "image://favicon/") ? icon.toString().substring(16) : icon)
                 }
             }
-
+            /*
             //onGeolocationPermissionRequested: requestGeolocationPermission(request)
 */
```

Comment:
Is there any reason for leaving this commented-out code in the file? If not, let's just remove it.